Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs

```diff
@@ -40,25 +58,32 @@ public static class WebApiServiceCollectionExtensions
             string configSectionName = "AzureAd",
             bool subscribeToJwtBearerMiddlewareDiagnosticsEvents = false)
         {
-            services.AddAuthentication(AzureADDefaults.JwtBearerAuthenticationScheme)
-                    .AddAzureADBearer(options => configuration.Bind(configSectionName, options));
-            services.Configure<AzureADOptions>(options => configuration.Bind(configSectionName, options));
+            services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
+                    .AddJwtBearer((options => configuration.Bind(configSectionName, options)));
+                  
+            services.Configure<MicrosoftIdentityOptions>(options => configuration.Bind(configSectionName, options));
 
             services.AddHttpContextAccessor();
 
             // Change the authentication configuration to accommodate the Microsoft identity platform endpoint (v2.0).
-            services.Configure<JwtBearerOptions>(AzureADDefaults.JwtBearerAuthenticationScheme, options =>
+            services.Configure<JwtBearerOptions>(JwtBearerDefaults.AuthenticationScheme, options =>
             {
+                var microsoftIdentityOptions = configuration.GetSection(configSectionName).Get<MicrosoftIdentityOptions>();
+
                 // Reinitialize the options as this has changed to JwtBearerOptions to pick configuration values for attributes unique to JwtBearerOptions
                 configuration.Bind(configSectionName, options);
 
-                // This is an Microsoft identity platform Web API
-                options.Authority += "/v2.0";
+                if (string.IsNullOrWhiteSpace(options.Authority))
+                    options.Authority = AuthorityHelpers.BuildAuthority(microsoftIdentityOptions);
+
+                if (!AuthorityHelpers.IsV2Authority(options.Authority))
+                    options.Authority += "/v2.0";
 
                 // The valid audiences are both the Client ID (options.Audience) and api://{ClientID}
                 options.TokenValidationParameters.ValidAudiences = new string[]
                 {
-                    options.Audience, $"api://{options.Audience}"
+                    // If the developer doesnt set the Audience on JwtBearerOptions, use ClientId from MicrosoftIdentityOptions
+                    options.Audience, $"api://{options.Audience ?? microsoftIdentityOptions.ClientId}"
```

Comment:
The `JwtBearerOptions` doesnt have `ClientId` as attribute, but `Audience`. So in case the developers dont include `Audience` in the appsettings, we can use `ClientId` from `MicrosoftIdentityOptions` as a fallback plan.